src/main/target/common_pre.h

```diff
@@ -62,6 +62,7 @@
 #define USE_MOTOR
 #define USE_DMA
 #define USE_TIMER
+#define USE_SERIAL_PIN_CONFIG
```

Comment:
I'd prefer to keep `USE_<something>` as something configurable.  SERIAL_PIN_CONFIG is not exactly configurable - it is property of target.
So maybe `SERIAL_TRAIT_PIN_CONFIG`, ot some other name that is not mixed with USE_

Also, it may be currently defined as `(defined(USE_SOFTSERIAL) || (defined(USE_UART) && !defined(SITL)) || defined(USE_LPUART))`. (UART is overloaded as TCP on SITL)

Reply:
We can start to use the trait model going forward for anything that is a part of the target, and leave USE_ for those items in config.h, and able to be selected by the user in the build process. I will setup some documentation on this approach in the development section.

Reply:
(is not specific only to target, for example pin_config is dependent on  USE_ configuration)